### PR TITLE
[clang] Fix unexpected warnings after a01307a

### PR DESCRIPTION
**Pull request by Fznamznon**
a01307a broke silencing of -Wmissing-field-initializers warnings in C for nested designators. This fixes the issue.

### CODE REVIEW

File: clang/lib/Sema/SemaInit.cpp [hunk 1]
```diff
@@ -864,6 +864,14 @@ InitListChecker::FillInEmptyInitializations(const InitializedEntity &Entity,
       WarnIfMissingField &=
           SemaRef.getLangOpts().CPlusPlus || !hasAnyDesignatedInits(SForm);
 
+      if (OuterILE) {
```
Comment (mikaelholmen) [inline clang/lib/Sema/SemaInit.cpp:867]: Perhaps add a comment here about why this is needed?

**Review by mikaelholmen — COMMENTED** [review]
I really don't know this, and I have no easy way to test that this really fixes all the new warnings we see with a01307a6ee788, but at least it seems to fix the added cases in the missing-field-initializers.c test so it looks like a step in the right direction.

File: clang/lib/Sema/SemaInit.cpp (hunk 1 above)
Comment (Fznamznon) [reply clang/lib/Sema/SemaInit.cpp:867]: Okay, sure.

Comment (mikaelholmen) [reply clang/lib/Sema/SemaInit.cpp:867]: Ok, ideally this should be reviewed by someone who knows the code but it LGTM.

**Review by mikaelholmen — APPROVED** [review]
Ok, ideally this should be reviewed by someone who knows the code but it LGTM.

**Comment by Fznamznon** [comment]
> Ok, ideally this should be reviewed by someone who knows the code but it LGTM.

I know, but a lot of folks are now on vacation already. The change is not complicated, so I'll probably wait till green pre-commit, then leave it for post-commit review to resolve the issue.

**Comment by mikaelholmen** [comment]
> > Ok, ideally this should be reviewed by someone who knows the code but it LGTM.
> 
> I know, but a lot of folks are now on vacation already. The change is not complicated, so I'll probably wait till green pre-commit, then leave it for post-commit review to resolve the issue.

Yeah, sounds good. Thanks!

**Comment by Fznamznon** [comment]
Ok, adding more reviewers. Please feel free to do post-commit review.

**Pull request merged**